sdk/cosmosdb/cosmos/src/request/request.ts

```diff
@@ -128,8 +128,13 @@ export async function getHeaders({
   }
 
   if (options.maxIntegratedCacheStalenessInMs && resourceType === ResourceType.item) {
-    headers[Constants.HttpHeaders.DedicatedGatewayPerRequestCacheStaleness] =
-      options.maxIntegratedCacheStalenessInMs;
+    if (typeof options.maxIntegratedCacheStalenessInMs === "number") {
+      headers[Constants.HttpHeaders.DedicatedGatewayPerRequestCacheStaleness] = String(
+        options.maxIntegratedCacheStalenessInMs
+      );
+    } else {
+      headers[Constants.HttpHeaders.DedicatedGatewayPerRequestCacheStaleness] = "null";
```

Comment:
Its silently ignoring and ambiguous to caller on when. 
One option is to error out with RangeError.